src/org/zaproxy/zap/extension/pscanrulesAlpha/FeaturePolicyScanner.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2019 The ZAP Development Team
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import java.util.Vector;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+import net.htmlparser.jericho.Source;
+
+/**
+ * Feature Policy Header Missing passive scan rule 
+ * https://github.com/zaproxy/zaproxy/issues/4885
+ */
+
+public class FeaturePolicyScanner extends PluginPassiveScanner {
+
+	private static final String MESSAGE_PREFIX = "pscanalpha.featurepolicymissing.";
+	private static final int PLUGIN_ID = 10063;
+	
+	private PassiveScanThread parent = null;
+	private static final Logger logger = Logger.getLogger(FeaturePolicyScanner.class);
```

Comment:
This could be moved up with the other constants and it’s name changed to full caps

Reply:
Resolved with 53ffa8ff6185fd0517d7c9a7862bcfce3109a226